.pre-commit-config.yaml

```diff
@@ -44,27 +21,17 @@ repos:
         args:
           - --autofix
 
-  - repo: https://github.com/PyCQA/flake8
-    rev: 7.1.1
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.9.2
     hooks:
-      - id: flake8
-        additional_dependencies: [
-          # flake8-blind-except, FIXME
-          flake8-builtins,
-          flake8-rst-docstrings,
-          # flake8-logging-format, FIXME
-        ]
-        args: [
-          # default black line length is 88
-          "--max-line-length=88",
-          # Conflicts with black: E203 whitespace before ':'
-          # Conflicts with PEP8 and black:
-          # W503 line break before binary operator
-          # Does not recognize deprecated directive in docstrings
-          "--ignore=E203,RST303,W503",
-          "--builtins-allowed-modules=csv,io,types",
-          "--rst-roles=class,func,ref,mod,meth,const",
-        ]
+    - id: ruff
+      args: ["--fix", "--show-fixes"]
+
+  - repo: https://github.com/psf/black
+    rev: 24.10.0
+    hooks:
+      - id: black
+        args: [--target-version=py36]
```

Comment:
should this not be 3.10 now?

Reply:
From the [`--target-version`](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html) documentation:
> By default, Black will infer target versions from the project metadata in `pyproject.toml`, specifically the `[project.requires-python]` field. If this does not yield conclusive results, Black will use per-file auto-detection.

I think this should be removed altogether...